crates/pathfinder/examples/dump_events.rs

```diff
@@ -86,24 +106,65 @@ fn main() -> anyhow::Result<()> {
     fs::create_dir_all(&cli.output_dir)?;
     let output_path = make_ground_path(&cli)?;
     let mut output_file = fs::File::create(output_path)?;
+
     let address = if let Some(addr) = cli.address {
         let felt = Felt::from_hex_str(&addr)?;
         Some(ContractAddress::new_or_panic(felt))
     } else {
         None
     };
+
+    let keys = if let Some(mut raw_keys) = cli.keys {
+        while let Some(tail) = raw_keys.chars().last() {
+            if tail == ',' {
+                raw_keys.pop();
+            } else {
+                break;
+            }
+        }
```

Comment:
This could be simplified using [trim_end_matches()](https://doc.rust-lang.org/std/string/struct.String.html#method.trim_end_matches)?

BTW, this trimming might actually be a bad thing. The filter does _not_ match events where the length of `keys` is longer than the number of values in this filter, so by trimming "empty" keys from the end of the filter we might be limiting what filters one can use. (For example: ",0x1," would match keys [0x0, 0x1, 0x2], but this trimming removes the last `,` so the filter effectively becomes ",0x1" which does _not_ match.)

Reply:
Hmm, you're right, indeed, it does match. I just misinterpreted the `if keys.len() > event.keys.len()` condition.